src/libraries/Microsoft.PowerFx.Core/Functions/UserDefinedFunction.cs

```diff
@@ -148,25 +158,59 @@ public override IEnumerable<StringGetter[]> GetSignatures()
             return new[] { _args.Select<UDFArg, TexlStrings.StringGetter>(key => _ => key.NameIdent.Name.Value).ToArray() };
         }
 
+        public (IntermediateNode topNode, ScopeSymbol ruleScopeSymbol) GetIRTranslator()
+        {
+            return IRTranslator.Translate(_binding);
+        }
+
+        public RecordValue GetRuntimeRecordValue(FormulaValue[] args, CancellationToken cancellationToken)
+        {
+            var argsArray = _args.ToArray();
```

Comment:
>var argsArray = _args.ToArray(); [](http://example.com/codeflow?start=12&length=32)

Any specific reason to change to array? 
you could use ElementAt() right?

Reply:
To avoid extra iterations by calling `.Count()`. Also, as far as I know `ElementAt` will iterate through the enumerable every time it is called.

Reply:
Why do we need to call the count? can't we foreach?

Reply:
Right, ElementAt is O[N].
The extra ToArray() usually isnt a problem. 

But if we're already creating a List<>, then we could just foreach and add to the list.